### ISSUE TITLE
box.execute should be immutable function 

### ISSUE THREAD

**Issue by R-omk** [tarantool/tarantool#4231]
Tarantool version:
version: 2.2.0-283-gd11b552e3

```
local box_execute = box.execute

box.cfg {
-- config
}

box_execute("SELECT 1")
```

it lead to error:
```
'builtin/box/load_cfg.lua:479: bad argument #1 to ''pairs'' (table expected, got  nil)'
```

https://github.com/tarantool/tarantool/blob/d11b552e3/src/box/lua/load_cfg.lua#L479

**Comment by Gerold103** [comment]
You do not have guarantees about box method before box.cfg is called. You can only count on box.cfg, that is all.

**Comment by kostja** [comment]
I asked @R-omk to report this since I believe the following should be done:
- we currently look at box.execute and change its type to implement automatic invocation of box.cfg{}. We should think of other way of doing this, not replacing box.execute()

**Comment by Gerold103** [comment]
Ok, you believe, but what are the reasons?

**Comment by Gerold103** [comment]
We can do that, just move current implementation of `box.execute` to something like `box.do_execute`, and inside `box.execute()` always call `do_execute`. The latter will be replaced on first `box.cfg`. But it is a perf penalty, obviously.

**Comment by kostja** [comment]
We can well use our own internal flag to look at, and not make the life of a user more difficult just because we can.

**Comment by kostja** [comment]
OK, first of all, we agreed a while ago that we should invoke box.cfg{} automatically and not just for SQL. So it's not only box.execute() that is affected. The automatic invocation as it is currently in 2.x is unfinished.

**Comment by Totktonada** [comment]
One more similar case:

```
tarantool> box_cfg = box.cfg
--- 
... 
tarantool> box_cfg()
<...>
tarantool> box_cfg()
--- 
- error: 'builtin/box/load_cfg.lua:19: builtin/box/load_cfg.lua:547: bad argument
    #1 to ''pairs'' (table expected, got nil)'
... 
```

**Comment by Totktonada** [comment]
[Cherry-picked](https://lists.tarantool.org/pipermail/tarantool-patches/2020-June/017944.html) box configuration fix for 1.10 and pushed it in [1.10.6-52-g9dd9b10f0](https://github.com/tarantool/tarantool/commit/9dd9b10f0047eeb31dea65615bef32577b22a4ae).
